src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.android.cs

```diff
@@ -22,18 +28,106 @@ public class MauiMediaElement : CoordinatorLayout
 	{
 		this.playerView = playerView;
+		this.context = context;
 
-		// Create a RelativeLayout for sizing the video
-		RelativeLayout relativeLayout = new(context)
+		playerView.FullscreenButtonClick += OnFullscreenButtonClick;
+		this.playerView.SetBackgroundColor(Android.Graphics.Color.Black);
+
+		AddView(playerView);
+	}
+
+	/// <summary>
+	/// Allows the video to enter or exist a full screen mode
+	/// </summary>
+	/// <param name="sender"></param>
+	/// <param name="e"></param>
+	void OnFullscreenButtonClick(object? sender, StyledPlayerView.FullscreenButtonClickEventArgs e)
+	{
+		// Ensure there is a player view
+		if (this.playerView is null)
+		{
+			return;
+		}
+
+		// Ensure current activity and window are available
+		if (Platform.CurrentActivity is not Activity currentActivity
+			|| currentActivity.Window is not Android.Views.Window currentWindow
+			|| currentActivity.Resources is not Resources currentResources
+			|| currentResources.Configuration is null)
+		{
+			return;
+		}
+
+		var windowInsetsControllerCompat = WindowCompat.GetInsetsController(currentWindow, currentWindow.DecorView);
+
+		if (e.IsFullScreen)
+		{
+			// Force the landscape on the device
+			currentActivity.RequestedOrientation = Android.Content.PM.ScreenOrientation.Landscape;
+
+			// Hide the SystemBars and Status bar
+			if (OperatingSystem.IsAndroidVersionAtLeast(30))
+			{
+				currentWindow.SetDecorFitsSystemWindows(false);
+
+				var windowInsets = currentWindow.DecorView.RootWindowInsets;
+				if (windowInsets is not null)
+				{
+					isSystemBarVisible = windowInsets.IsVisible(WindowInsetsCompat.Type.NavigationBars()) || windowInsets.IsVisible(WindowInsetsCompat.Type.StatusBars());
+					
+					if (isSystemBarVisible)
+					{
+						windowInsetsControllerCompat.SystemBarsBehavior = WindowInsetsControllerCompat.BehaviorShowTransientBarsBySwipe;
+						currentWindow.InsetsController?.Hide(WindowInsets.Type.SystemBars());
+					}
+				}
+			}
+			else
+			{
+				defaultSystemUiVisibility = (int)currentWindow.DecorView.SystemUiVisibility;
+				int systemUiVisibility = defaultSystemUiVisibility | (int)SystemUiFlags.LayoutStable | (int)SystemUiFlags.LayoutHideNavigation | (int)SystemUiFlags.LayoutHideNavigation |
+					(int)SystemUiFlags.LayoutFullscreen | (int)SystemUiFlags.HideNavigation | (int)SystemUiFlags.Fullscreen | (int)SystemUiFlags.Immersive;
+				currentWindow.DecorView.SystemUiVisibility = (StatusBarVisibility)systemUiVisibility;
+			}
+
+			// Update the PlayerView
+			if (currentWindow.DecorView is FrameLayout layout)
+			{
+				RemoveView(playerView);
+				layout.AddView(playerView, new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent));
+			}
+		}
+		else
 		{
-			LayoutParameters = new CoordinatorLayout.LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent)
+			currentActivity.RequestedOrientation = Android.Content.PM.ScreenOrientation.Portrait;
```

Comment:
Should we be caching this value before we go into `IsFullScreen == true` so that we only reset it back if it needs to be?

Reply:
Thanks @bijington for your feedback. This is something up to bring to discussion in later phases, because the expectation of the behavior of the Full screen on mobile for example is that if you are in portrait mode, the video goes to full screen in landscape mode and if they cancel the full screen then it goes back to portrait mode. 

However, if the app is on a tablet and you have it oriented in landscape mode, when you return back from full screen it will maintain the state, that is what happened in our results. This is how the YouTube app works for mobile and table.

I believe we can do this way right now, and once we start getting feedback we can learn and improve over the time if needed.

What do you think?

Reply:
If the correct orientation is maintained even if we are requesting portrait then that is good. Strange but good :)

Out of interest have you tested the full screen mode in an application that doesn't support landscape orientation? Does that behave?

Reply:
Yes, it maintains the portrait mode. If you block your app on an orientation it will respect the orientation. We tested that particular case on an Android tablet.

Reply:
With this code ``` urrentActivity.RequestedOrientation = Android.Content.PM.ScreenOrientation.Landscape;  ```
or ``` urrentActivity.RequestedOrientation = Android.Content.PM.ScreenOrientation.Portrait; ```
it appears once set it becomes locked. This is unexpected behavior and possibly a bug? Why we don't know. But auto rotation becomes disabled if either is set anywhere in code. I talked to @vhugogarcia and we decided to remove it to restore default rotation behavior back. 

I tested changes with removal of above and auto rotation now works as expected.